transformer_engine/common/common.h

```diff
@@ -154,12 +155,10 @@ struct Tensor {
     return acc;
   }
 
-  bool has_data() const noexcept { return data.dptr != nullptr; }
+  bool has_data() const noexcept { return data.has_data(); }
 
   // Check for size (not just pointer) for 0-dim or no token cases.
-  bool has_columnwise_data() const noexcept {
-    return columnwise_data.dptr != nullptr || columnwise_data.shape.size() != 0;
-  }
+  bool has_columnwise_data() const noexcept { return columnwise_data.has_data(); }
```

Comment:
This is a nice approach. If this causes issues or test failures, I propose we merge https://github.com/NVIDIA/TransformerEngine/pull/2330 first rather than trying to fix them in this PR.